TemporalConvolution.lua

```diff
@@ -87,6 +86,11 @@ function TemporalConvolution:updateGradInput(input, gradOutput)
    if not self.gradInput then return end
    local _gradOutput = transposeGradOutput(gradOutput,self.buffer)
    local _input = inputview(input)
+   if input:dim()==3 and self.gradInput:dim() == 3 then
```

Comment:
I think instead of applying views here, it is better to check if gradInput:dim() == 4 after updateGradInput, and then apply view to it.